SparseGrids/TasmanianSparseGridWrapC.cpp

```diff
@@ -478,6 +478,56 @@ void tsgGetGPUName(int gpu, int num_buffer, char *buffer, int *num_actual){
 
 void tsgDeleteInts(int *p){ delete[] p; }
 
+void* tsgConstructCustomTabulated(){ return (void*) new CustomTabulated(); }
+void tsgDestructCustomTabulated(void* ct){ delete ((CustomTabulated*) ct); }
+
+void tsgWriteCustomTabulated(void *ct, const char* filename, const bool use_binary_format){
+    std::ofstream ofs(filename, std::ios::out);
+    if (use_binary_format) {
+        ((CustomTabulated*) ct)->write<false>(ofs);
+    } else {
+        ((CustomTabulated*) ct)->write<true>(ofs);
+    }
+}
+int tsgReadCustomTabulated(void *ct, const char* filename){
+    try{
+        ((CustomTabulated*) ct)->read(filename);
+        return 1;
+    }catch(std::runtime_error &e){
+        cerr << e.what() << endl;
+        return 0;
+    }
+}
+
+int tsgGetNumLevelsCustomTabulated(void* ct){ return ((CustomTabulated*) ct)->getNumLevels(); }
+int tsgGetNumPointsCustomTabulated(void* ct, const int level){ return ((CustomTabulated*) ct)->getNumPoints(level); }
+int tsgGetIExactCustomTabulated(void* ct, const int level){ return ((CustomTabulated*) ct)->getIExact(level); }
+int tsgGetQExactCustomTabulated(void* ct, const int level){ return ((CustomTabulated*) ct)->getQExact(level); }
+const char* tsgGetDescriptionCustomTabulated(void* ct) { return ((CustomTabulated*) ct)->getDescription(); }
+
+void tsgGetWeightsNodesStaticCustomTabulated(void* ct, int level, double* w, double* x) {
+    int num_points = ((CustomTabulated*) ct)->getNumPoints(level);
+    std::vector<double> vec_w(num_points), vec_x(num_points);
+    ((CustomTabulated*) ct)->getWeightsNodes(level, vec_w, vec_x);
```

Comment:
Update `CustomTabulated` to have an overload of `getWeightsNodes()` that directly write to arrays, no need to use vectors here and do an extra copy.

Reply:
Done.